tests/utils/test_settings_utils.py

```diff
@@ -0,0 +1,164 @@
+import json
+import os
+from pathlib import Path
+from unittest import mock
+from django.test import TestCase, override_settings
+from django.conf import settings
+from arches.settings_utils import generate_frontend_configuration
+
+
+# these tests can be run from the command line via
+# python manage.py test tests.utils.test_settings_utils --settings="tests.test_settings"
+
+
+class TestFrontendConfigurationGeneration(TestCase):
+    """
+    Test the generation of frontend configuration files.
+    """
+
+    @mock.patch("arches.settings_utils.open")
+    @mock.patch("arches.settings_utils.json.dump")
+    @mock.patch("arches.settings_utils.list_arches_app_names")
+    @mock.patch("arches.settings_utils.list_arches_app_paths")
+    @mock.patch("arches.settings_utils.os.path.realpath")
+    def test_generate_frontend_configuration_core_arches(
+        self, mock_realpath, mock_app_paths, mock_app_names, mock_json_dump, mock_open
+    ):
+        """
+        Test frontend configuration generation when running core arches directly.
+        """
+        # Mock settings
+        mock_settings = mock.MagicMock()
+        mock_settings.APP_ROOT = "/arches/app"
+        mock_settings.ROOT_DIR = "/arches"
+        mock_settings.PUBLIC_SERVER_ADDRESS = "http://localhost:8000"
+        mock_settings.STATIC_URL = "/static/"
+        mock_settings.WEBPACK_DEVELOPMENT_SERVER_PORT = 8080  # Add this missing setting
+
+        # Mock path responses
+        mock_realpath.side_effect = lambda path: path
+
+        # App names and paths mocks
+        mock_app_names.return_value = ["arches", "test_app"]
+        mock_app_paths.return_value = ["/arches/path", "/test/app/path"]
+
+        # Execute with mocked settings
+        with mock.patch("arches.settings_utils.settings", mock_settings):
+            generate_frontend_configuration()
+
+        # Check file paths
+        mock_open.assert_any_call(
+            "/arches/../.frontend-configuration-settings.json", "w"
+        )
+        mock_open.assert_any_call("/arches/../.tsconfig-paths.json", "w")
+
+        # Check content of first file
+        expected_settings_data = {
```

Comment:
I wouldn't be afraid of removing much of the mocking and writing this file to a tempdir, but I leave that to you.

Reply:
I've stripped most out now